ggshield/scan/scannable.py

```diff
@@ -140,8 +141,14 @@ def apply_filter(self, filter_func: Callable[[File], bool]) -> "Files":
 
     def get_tar_stream(self) -> BytesIO:
         tar_stream = BytesIO()
+        current_dir_size = 0
         with tarfile.open(fileobj=tar_stream, mode="w:gz") as tar:
             for filename in self.files:
+                current_dir_size += os.path.getsize(filename)
+                if current_dir_size > MAX_TOTAL_FILES_SIZE:
+                    raise click.ClickException(
+                        "The total size of the files processed exceeds 30MB, please try again with less files"
```

Comment:
The message should reuse the constant (but divided by 1024 * 1024) to ensure it does not get out of sync.